op-node/rollup/derive/frame.go

```diff
@@ -80,6 +81,7 @@ func (f *Frame) UnmarshalBinary(r ByteReader) error {
 		return fmt.Errorf("error reading ID time: %w", err)
 	}
 	// stop reading and ignore remaining data if we encounter a zeroed ID
+	// TODO: this is probably incorrect
```

Comment:
Why is this probably incorrect?

Reply:
I don't think we spec out that if `f.ID` is empty we claim the channel is empty. I also think this is a hold over from some old code which would use this as an end condition (i.e. could no longer read from the data)

Reply:
The empty channel ID check is a hold-over from old code, to allow zeroed data to work as valid padding. It's fine to remove, not really applicable until we deal with fixed-size data (like eip 4844)